contrib/objectstore/subscriber/objectstore.go

```diff
@@ -54,45 +76,104 @@ func (s *Subscriber) OnStructMessage(c ws.Speaker, msg *ws.StructMessage) {
 	}
 }
 
-// StoreFlows writes a set of flows to the object storage service
+// StoreFlows store flows in memory, before being written to the object store
 func (s *Subscriber) StoreFlows(flows []*flow.Flow) error {
-	if len(flows) == 0 {
-		return nil
+	endTime := time.Now()
+
+	s.flowsMutex.Lock()
```

Comment:
can you explain why is the mutex needed?

Reply:
because I call `StoreFlows` asynchronously, using a `time.AfterFunc`.

Reply:
ok